db/memtable.cc

```diff
@@ -960,53 +963,58 @@ void MemTable::MultiGet(const ReadOptions& read_options, MultiGetRange* range,
   }
   PERF_TIMER_GUARD(get_from_memtable_time);
 
+  // For now, memtable Bloom filter is effectively disabled if there are any
+  // range tombstones. This is the simplest way to ensure range tombstones are
+  // handled. TODO: allow Bloom checks where max_covering_tombstone_seq==0
+  bool no_range_del = read_options.ignore_range_deletions ||
+                      is_range_del_table_empty_.load(std::memory_order_relaxed);
   MultiGetRange temp_range(*range, range->begin(), range->end());
-  if (bloom_filter_) {
-    std::array<Slice*, MultiGetContext::MAX_BATCH_SIZE> keys;
-    std::array<bool, MultiGetContext::MAX_BATCH_SIZE> may_match = {{true}};
-    autovector<Slice, MultiGetContext::MAX_BATCH_SIZE> prefixes;
+  if (bloom_filter_ && no_range_del) {
```

Comment:
You want to skip all filter checks when any range tombstone is present? I thought in the previous version of this PR it only skipped checking for a given lookup key when a range tombstone covering that particular lookup key was found.

Reply:
Never mind, it was always like that. I guess you succeeded at "Making that more clear in the code"

Reply:
> You want to skip all filter checks when any range tombstone is present?

I think that's best for now (backport-able with low risk)

> I thought in the previous version ...

Looks can be deceiving. If you dig into details, `range_del_iter == nullptr` is logically equivalent to `no_range_del`. I believe you'd have to check `max_covering_tombstone_seq==0` for coverage check, but I'm not super confident and that's more than I want to write unit tests for at the moment.

Reply:
Sorry let me start from the beginning. I originally wondered why we skip filter at all when there are range tombstones. They seem useful whether we found a range tombstone or not. Perhaps that can be explained in the comment/TODO -- or if we don't need to skip, perhaps it can give the optimization without introducing edge cases to test.